src/persistence/history.cpp

```diff
@@ -43,23 +43,33 @@ MessageState getMessageState(bool isPending, bool isBroken)
     return messageState;
 }
 
-QString generatePeerIdString(ToxPk const& pk)
+QString generateAuthorIdString()
 {
-    return QString("(SELECT id FROM peers WHERE public_key = '%1')").arg(pk.toString());
+    return QString("(SELECT id FROM authors WHERE public_key = ?)");
```

Comment:
Not a new question, but what happens if you accidentally add the same PK twice?

Reply:
You would get an SQL error on insert that would reject the insert, due to `public_key` being marked as `UNIQUE` in the schema.

In general for sub-queries though, I'm not sure?